samples/javascript_nodejs/57.teams-conversation-bot/bots/teamsConversationBot.js

```diff
@@ -31,42 +32,36 @@ class TeamsConversationBot extends TeamsActivityHandler {
             } else {
                 await this.cardActivityAsync(context, false);
             }
+
+            await next();
         });
 
         this.onMembersAddedActivity(async (context, next) => {
-            //  console.log(context);
-            context.activity.membersAdded.forEach(async (teamMember) => {
-                if (teamMember.id !== context.activity.recipient.id) {
+            await Promise.all((context.activity.membersAdded || []).map(async (member) => {
```

Comment:
Calling `.forEach` with an async function is fine, but there's no guarantee that when the `forEach` invocation is done that all the async functions are done. `Promise.all` must be used to synchronize.